Grabbers can now only contain things smaller than the assembly.

DaxDupont opened the pull request:
[Changelogs]: 

:cl: Dax Dupont
balance: Grabbers/throwers no longer can contain/throw things equal to the assembly size. 
/:cl:

[why]: I think this was a logic error.
Grabbers can hold 10 items, so arguably the objects shouldn't be the same weight class as the assembly holding the object.

DaxDupont committed:
Grabbers can now only contain things smaller than the assembly

arsserpentarium commented:
ok

DaxDupont committed:
Good suggestion buggy

DaxDupont commented:
actually never mind they should be allowed to throw things that are larger.

DaxDupont committed:
fucking github broke

evsey9 commented:
Aaaactuaally I repeatedly stated in my PR that this WAS NOT a logic issue.

evsey9 commented:
Also this ruins all small circuit bots that use grabbers by making them completely useless

arsserpentarium commented:
At first autothrowing of assemblies was a bug. But I found it hilarious and leaved.
If you do this, barely you will make some deathmachine harder to make. But will remove fun.

DaxDupont commented:
Small circuits should not be able to hold 10 small items

On Feb 9, 2018 11:30, "arsserpentarium" <notifications@github.com> wrote:

> At first autothrowing of assemblies was a bug. But I found it hilarious
> and leaved.
> If you do this, barely you will make some deathmachine harder to make. But
> will remove fun.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/tgstation/tgstation/pull/35462#issuecomment-364396473>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AQcGeBcFQezvy4izTGhCdVkfNEex3iZJks5tTB4vgaJpZM4R-2Fz>
> .
>

DaxDupont commented:
They can still hold tiny items

On Feb 9, 2018 11:53, "Dax Dupont" <skyemenjou@gmail.com> wrote:

> Small circuits should not be able to hold 10 small items
>
> On Feb 9, 2018 11:30, "arsserpentarium" <notifications@github.com> wrote:
>
>> At first autothrowing of assemblies was a bug. But I found it hilarious
>> and leaved.
>> If you do this, barely you will make some deathmachine harder to make.
>> But will remove fun.
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/tgstation/tgstation/pull/35462#issuecomment-364396473>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe-auth/AQcGeBcFQezvy4izTGhCdVkfNEex3iZJks5tTB4vgaJpZM4R-2Fz>
>> .
>>
>

arsserpentarium commented:
I am not about grabbers. I think,if grabbers will store only smaller things,it's ok.
I was talking about throwers

DaxDupont commented:
@evesy
Hatchets and this are related. 
Small circuits can still store tiny items. 

The problem was that circuits could store 10 hatchets and throw them all at once.
We should not have some magical storage that can hold 10 same weight items.

Circuits aren't hard to get when you literally copy paste the code from the forums or reddit.

I'll concede that this is balance because originally I didn't expect the original coder to be retarded

DaxDupont commented:
If you want to add bluespess grabbers that's fine, as long they hold equal weight or whatever.

I'd make that a separate thing so grabbers don't become insanely expensive for other purposes.

DaxDupont commented:
Also I'd like to add I did this before I saw you the changelog of your PR.

Tlaltecuhtli commented:
why does it have a fix tag

DaxDupont commented:
because it doesn't update automatically

ACCount12 commented:
That's not a fix, that's an "I DED"-tier balance meme. @arsserpentarium wrote both thrower and grabber, and he says it was intentional. I rewrote both thrower and grabber, and I say this was intentional.

Grabber may need some nerfs, but it's hard to abuse it, so I'd say it's fine as it is.

DaxDupont commented:
I changed the tag in the changelog, please read. Also it's easy to abuse
when there's a ready made circuit on the forums

On Feb 10, 2018 02:25, "ACCount" <notifications@github.com> wrote:

> That's not a fix, that's an "I DED"-tier balance meme. @arsserpentarium
> <https://github.com/arsserpentarium> wrote both thrower and grabber, and
> he says it was intentional. I rewrote both thrower and grabber, and I say
> this was intentional.
>
> Grabber may need some nerfs, but it's hard to abuse it, so I'd say it's
> fine as it is.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/tgstation/tgstation/pull/35462#issuecomment-364614063>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AQcGeAUjSKk1Q_J8sTDMeGSPVPPTra9nks5tTPAZgaJpZM4R-2Fz>
> .
>

arsserpentarium commented:
Still tag "i ded" is required.

DaxDupont commented:
I don't mess with circuits much but when I saw people running around with really weird things and then I see the forum post that says it instacrit people I went to find out what the fuck is wrong with it.

evsey9 commented:
perhaps its throwers that needs a nerf and not grabbers?
you can still instacrit people even with this PR merged lmao

DaxDupont commented:
Yeah with bigger bots.

Though that's an issue on it's own. Circuits don't become bags of holding for some reason.
I am 100% okay with there being a variant that consumes bluespess mesh.

Also with the throwers, I'd like some input on how to fix/balance them. Considering the issue lies with them firing at the same time, and not the delay as I previously thought.

evsey9 commented:
circuits don't become bags of holding, they don't go onto your belt, backpack or pocket slots ( the bigger ones )
also normal sized circuits still can do that as far as i know

DaxDupont commented:
They can fit small or tiny yes,
Bulky can fit normal/small/tiny.

Like I said, I am still looking for something that prevents throwers from causing issues mass throwing, but I am not sure how to do that without the raising the complexity but that's not a good option.

DaxDupont commented:
@arsserpentarium I would like your input on how to deal with throwers being able to toss all of the contents at once.

arsserpentarium commented:
ideas:
1.cooldown 0.1 s for extenal manipulation. Or cap on external activity per tick. So you just can't fire them all at once.
2.make circuit require capacitor. Which limits power of assembly.So 10 throwers at once hits power cap.
3.limit slots or external manipulation;make these slots to be "ports" on circuit case sprite.And for every manipulative circuit make their appearance from that port.
Would be cool, because provides custom sprites as well, but too complex to make, maintain and play.

ShizCalev commented on code/modules/integrated_electronics/subtypes/manipulation.dm:
```diff
@@ -315,16 +315,10 @@
 		if(mode == 1)
 			if(check_target(AM))
 				var/weightcheck = FALSE
-				if ((!istype(AM,/obj/item/device/electronic_assembly/)) && (!istype(AM,/obj/item/device/transfer_valve)))
```
should probably keep the transfer valve check
